### ISSUE TITLE
[web component] User Menu

### ISSUE THREAD

**Issue by gfellerph** [swisspost/design-system#5195]
### ♿ Accessibility

- List of user links should be keyboard accessible by pressing up and down arrows when the menu is open.
- Make sure the menu trigger has a valid accessible name
- Menu trigger and login should have a tooltip when no accessible name is being displayed

### 🧪 Testing

Test user specific components. Don't test already existing and tested components like `<post-menu>`.

### 💻 Proposed implementation

See https://github.com/swisspost/design-system/issues/5239.

```html
<!-- Logged out -->
<a href="/login">
  <post-icon name=login></post-icon>
  <span>Login</span>
</a>

<!-- Logged in -->

<!-- Trigger -->
<post-menu-trigger for=profile>
  <post-avatar>{user.name}</post-avatar>
</post-menu-trigger>

<!-- Dropdown -->
<post-menu id=profile>
  <div class="avatar-with-name">
    <post-avatar>{user.name}</post-avatar>
    <span>{user.name}</span>
  </div>
  <post-list>
    <post-list-link>
      <post-icon name=user></post-icon>
      <span>{user.go_to_profile}</span>
    </post-list-link>
    <!-- ... -->
  </post-list>
</post-menu>
```

### 📃 Tasks

- [x] [Naming TBD](https://teams.microsoft.com/l/message/19:meeting_YTYzYzA0YWQtNTQ1Ny00OWU3LTk1NGQtZTQ5YThhNTUwNWI3@thread.v2/1744819058002?context=%7B%22contextType%22%3A%22chat%22%7D)
- [ ] Implement component according to design
- [ ] Create storybook documentation
- [ ] Add functional tests
- [ ] Add integration tests for React
- [ ] Add integration tests for Angular
- [ ] Add visual regression tests
- [ ] Design review

**Comment by alizedebray** [comment]
**Update**: The user menu will not be developed as a web component. 
We will: 
- Add styles to the `post-menu` to make it a real component instead of a raw component
- Add parts to the `post-menu` so that its default style can be updated when needed
- Provide an example of user menu build from the post-menu

**Comment by alizedebray** [comment]
According to the discussion concerning prop drilling, the user menu will be implemented as a HTML/CSS component.
